### ISSUE TITLE
support for embedded YouTube videos

### ISSUE THREAD

**Issue by rsubr** [secure-77/Perlite#121]
Thanks for this wonderful project. I am effortlessly [publishing my notes online](https://github.com/secure-77/Perlite/issues/100#issuecomment-1774935499) in docker  using [Gitea webhooks](https://docs.gitea.com/usage/webhooks).

Obsidian supports [embedding YouTube videos]( 
https://help.obsidian.md/Editing+and+formatting/Embed+web+pages#Embed%20a%20YouTube%20video) using the same images markdown syntax. Perlite does not support this feature and incorrectly renders YouTube videos as images - none of the YouTube videos are loading. Obsidian Publish renders YouTube links correctly.

### Obsidian YouTube Markdown

```md
![Making a Minecraft Obsidian Block in Real Life Using Astro Tech Resin](https://www.youtube.com/watch?v=NnTvZWp5Q7o)
```

### Incorrect Perlite Rendering as IMG
Perlite does not support this and YouTube videos are rendered using an `img` tag like this:

```html
<img src="https://www.youtube.com/watch?v=NnTvZWp5Q7o"
    alt="Making a Minecraft Obsidian Block in Real Life Using Astro Tech Resin"
    class="external-link perlite-external-link" target="_blank" rel="noopener noreferrer">
```

### Obsidian Publish Rendering as IFRAME
The html used by `Obsidian Publish` for embedding YouTube is this:

```html
<iframe class="external-embed mod-receives-events"
    sandbox="allow-forms allow-presentation allow-same-origin allow-scripts allow-modals allow-popups"
    allow="fullscreen"
    frameborder="0"
    src="https://www.youtube.com/embed/NnTvZWp5Q7o">
</iframe>
```

Please note that in Obsidian also recognizes the below forms and the iframe is generated correctly:

```md
![](https://www.youtube.com/watch?v=NnTvZWp5Q7o)

![](https://www.youtu.be/NnTvZWp5Q7o)

![](https://youtu.be/NnTvZWp5Q7o)
```

Can we please get support for Obsidian YouTube links?

Thanks again for the great project!

**Comment by secure-77** [comment]
Thanks for the feedback and this well prepared issue! ❤️

This should be doable 👍 I will try to implement it for the next release!

**Comment by secure-77** [comment]
implemented in release [1.5.9](https://github.com/secure-77/Perlite/releases/tag/1.5.9)
